### PR TITLE
Update build_and_unit_test.yml for gpbackup workflow

### PR DESCRIPTION
**Pull request by Zhangbaowen-Hashdata**
---

## Change logs
To automate gpbackup's compilation and testing, we need to establish an efficient process that compiles and tests its code without human intervention.

## Contributor's checklist

Here are some reminders before you submit your pull request:

* Make sure that your Pull Request has a clear title and commit message. You can take the [Git commit template](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) as a reference.
* Sign the Contributor License Agreement as prompted for your first-time contribution (*One-time setup*).
* Learn the [code contribution](https://cloudberrydb.org/contribute/code) and [doc contribution](https://cloudberrydb.org/contribute/doc) guides for better collaboration.
* List your communications in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb-site/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
* Feel free to ask for the cloudberrydb team to help review and approve.

### CODE REVIEW

**Review by my-ship-it — APPROVED** [review]
LGTM

**Comment by my-ship-it** [comment]
Please add more detailed commit message?

**Comment by Zhangbaowen-Hashdata** [comment]
> Please add more detailed commit message?

OK ,i will make it

File: .github/workflows/build_and_unit_test.yml
Comment (tuhaihe) [inline .github/workflows/build_and_unit_test.yml:1]: Could we update the name of this workflow to make it more straightforward?
like:
> build_and_unit_test ~>  gpbackup_xxxxx ?

**Review by tuhaihe — APPROVED** [review]
LGTM.

**Pull request merged**